File: sherpa-onnx/java-api/src/com/k2fsa/sherpa/onnx/WaveWriter.java
// Copyright 2024 Xiaomi Corporation

package com.k2fsa.sherpa.onnx;

public class WaveWriter {
    public WaveWriter() {
    }

    public static boolean write(String filename, float[] samples, int sampleRate) {
        WaveWriter w = new WaveWriter();
        return w.writeWaveToFile(filename, samples, sampleRate);
    }

    private native boolean writeWaveToFile(String filename, float[] samples, int sampleRate);
}

File: flutter/sherpa_onnx/lib/src/web/wave_writer.dart
// Copyright (c)  2026  Xiaomi Corporation
// Web stub for WAV writing -- not yet implemented.
// TODO: implement using dart:js_interop and the corresponding JS wrapper.

import 'dart:typed_data';

/// Write normalized mono PCM samples to a WAV file.
///
/// Returns `true` on success and `false` otherwise.
bool writeWave(
    {required String filename,
    required Float32List samples,
    required int sampleRate}) {
  throw UnsupportedError('writeWave is not yet supported on web');
}
